transport-classes-io_uring/src/main/java/io/netty/channel/uring/AbstractIoUringStreamChannel.java

```diff
@@ -597,13 +639,158 @@ boolean writeComplete0(byte op, int res, int flags, short data, int outstanding)
             return true;
         }
 
+        private boolean handleSendZC(int op, int res, int flags,
+                                     IoUringSendZCMessage currentMessage, ChannelOutboundBuffer channelOutboundBuffer) {
+            if (res == Native.ERRNO_ECANCELED_NEGATIVE) {
+                // The sendzc is cancelled,so we should remove it from the queue.
+                if (op == Native.IORING_OP_SEND_ZC) {
+                    zcSendQueue.removeLast();
```

Comment:
why `removeLast()` ? Should this call `remove()` ?

Reply:
The ioUringOps being canceled is the most recent sendzc, which sits at the tail of the queue. 
When multiple sendzc requests are awaiting ACKs, it’s the latest one that gets canceled, so we use removeLast.